### ISSUE TITLE
Build fails with openssl-1.1

### ISSUE THREAD

**Issue by Polynomial-C** [TelepathyIM/telepathy-gabble#10]
```
/bin/sh ../libtool  --tag=CC   --mode=compile x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I..    -Wall -Wextra -Wdeclaration-after-statement -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wsign-compare -Wnested-externs -Wpointer-ari
th -Wformat-security -Winit-self -Wno-missing-field-initializers -Wno-unused-parameter  -pthread -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/libxml2 -pthread -I/usr/include/libsoup-2.4 -I/usr/include/libxml2 -I/u
sr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -DG_LOG_DOMAIN=\"wocky\" -DWOCKY_COMPILATION -march=native -mtune=native -O2 -pipe -c -o wocky-openssl.lo wocky-openssl.c
libtool: compile:  x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I.. -Wall -Wextra -Wdeclaration-after-statement -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wsign-compare -Wnested-externs -Wpointer-arith -Wformat-security -Winit-se
lf -Wno-missing-field-initializers -Wno-unused-parameter -pthread -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/libxml2 -pthread -I/usr/include/libsoup-2.4 -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib6
4/glib-2.0/include -DG_LOG_DOMAIN=\"wocky\" -DWOCKY_COMPILATION -march=native -mtune=native -O2 -pipe -c wocky-openssl.c  -fPIC -DPIC -o .libs/wocky-openssl.o
wocky-openssl.c: In function ‘check_peer_name’:
wocky-openssl.c:888:23: error: dereferencing pointer to incomplete type ‘X509’ {aka ‘struct x509_st’}
   X509_CINF *ci = cert->cert_info;
                       ^~
wocky-openssl.c:909:19: error: dereferencing pointer to incomplete type ‘X509_CINF’ {aka ‘struct x509_cinf_st’}
   if (!rval && (ci->extensions != NULL))
                   ^~
wocky-openssl.c:918:22: error: dereferencing pointer to incomplete type ‘X509_EXTENSION’ {aka ‘struct X509_extension_st’}
         int len = ext->value->length;
                      ^~
wocky-openssl.c: In function ‘_cert_status’:
wocky-openssl.c:1122:33: error: dereferencing pointer to incomplete type ‘X509_STORE’ {aka ‘struct x509_store_st’}
           long old_flags = store->param->flags;
                                 ^~
wocky-openssl.c: In function ‘wocky_tls_session_init’:
wocky-openssl.c:1681:31: warning: implicit declaration of function ‘CRYPTO_malloc_init’; did you mean ‘CRYPTO_malloc’? [-Wimplicit-function-declaration]
       malloc_init_succeeded = CRYPTO_malloc_init ();
                               ^~~~~~~~~~~~~~~~~~
                               CRYPTO_malloc
wocky-openssl.c:1681:31: warning: nested extern declaration of ‘CRYPTO_malloc_init’ [-Wnested-externs]
wocky-openssl.c: In function ‘wocky_tls_session_constructed’:
wocky-openssl.c:1781:7: warning: ‘TLSv1_server_method’ is deprecated [-Wdeprecated-declarations]
       session->method = (SSL_METHOD *) TLSv1_server_method ();
       ^~~~~~~
In file included from /usr/include/openssl/opensslconf.h:10,
                 from /usr/include/openssl/e_os2.h:13,
                 from /usr/include/openssl/ssl.h:15,
                 from wocky-openssl.c:47:
/usr/include/openssl/ssl.h:1853:1: note: declared here
 DEPRECATEDIN_1_1_0(__owur const SSL_METHOD *TLSv1_server_method(void))
 ^~~~~~~~~~~~~~~~~~
wocky-openssl.c:1786:7: warning: ‘TLSv1_client_method’ is deprecated [-Wdeprecated-declarations]
       session->method = (SSL_METHOD *) TLSv1_client_method ();
       ^~~~~~~
In file included from /usr/include/openssl/opensslconf.h:10,
                 from /usr/include/openssl/e_os2.h:13,
                 from /usr/include/openssl/ssl.h:15,
                 from wocky-openssl.c:47:
/usr/include/openssl/ssl.h:1854:1: note: declared here
 DEPRECATEDIN_1_1_0(__owur const SSL_METHOD *TLSv1_client_method(void))
 ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib.h:62,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from wocky-tls.h:32,
                 from wocky-openssl.c:40:
wocky-openssl.c: In function ‘wocky_tls_output_stream_write_async’:
/usr/include/glib-2.0/glib/gmessages.h:337:25: warning: this statement may fall through [-Wimplicit-fallthrough=]
 #define g_warning(...)  g_log (G_LOG_DOMAIN,         \
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                G_LOG_LEVEL_WARNING,  \
                                ~~~~~~~~~~~~~~~~~~~~~~~
                                __VA_ARGS__)
                                ~~~~~~~~~~~~
wocky-openssl.c:1369:11: note: in expansion of macro ‘g_warning’
           g_warning ("write caused read: unsupported TLS re-negotiation?");
           ^~~~~~~~~
wocky-openssl.c:1373:9: note: here
         default:
         ^~~~~~~
In file included from /usr/include/glib-2.0/glib.h:62,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from wocky-tls.h:32,
                 from wocky-openssl.c:40:
wocky-openssl.c: In function ‘ssl_read_is_complete’:
/usr/include/glib-2.0/glib/gmessages.h:337:25: warning: this statement may fall through [-Wimplicit-fallthrough=]
 #define g_warning(...)  g_log (G_LOG_DOMAIN,         \
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                G_LOG_LEVEL_WARNING,  \
                                ~~~~~~~~~~~~~~~~~~~~~~~
                                __VA_ARGS__)
                                ~~~~~~~~~~~~
wocky-openssl.c:533:11: note: in expansion of macro ‘g_warning’
           g_warning ("read caused write: unsupported TLS re-negotiation?");
           ^~~~~~~~~
wocky-openssl.c:537:9: note: here
         default:
         ^~~~~~~
make[8]: *** [Makefile:780: wocky-openssl.lo] Error 1
```

See also downstream bug report: https://bugs.gentoo.org/658902

**Comment by roelaaij** [comment]
PR submitted to wocky with suggested fix: TelepathyIM/wocky#1

**Comment by devurandom** [comment]
See-Also: TelepathyIM/telepathy-salut#1
